src/api/app/services/notification_service/notifier.rb

```diff
@@ -62,32 +62,25 @@ def call
         next if channel == :rss && @event.eventtype.in?(REJECTED_FOR_RSS)
 
         @event.subscriptions(channel).each do |subscription|
-          create_notification_per_subscription(subscription, channel)
+          create_notification(subscription, channel)
         end
       end
     end
 
     private
 
-    def create_notification_per_subscription(subscription, channel)
-      return unless create_notification?(subscription.subscriber, channel)
+    def create_notification(subscription, channel)
+      # conditions to pass to create the new event notification
+      return if subscription.subscriber.nil? || subscription.subscriber.away? || (channel == :rss && subscription.subscriber.rss_secret.blank?)
```

Comment:
You don't have to chain those with OR right?

```
return if a
return if b
return unless c
```
is way more readable.

Reply:
On one hand having one line for each _scoped check_ is nice (one for the subscriber conditions, one for the notifiable and one for the report). But splitting this up is even better. I'm adapting it, thanks